procinterrupts.c

```diff
@@ -143,6 +143,36 @@ static void guess_arm_irq_hints(char *name, struct irq_info *info)
 	
 	
 }
+
+static int is_arm_irq_aff_cannot_change(int irq)
+{
+	char buf[PATH_MAX] = { 0 };
+	FILE *file = NULL;
+	char *line = NULL;
+	size_t size = 0;
+	int ret = 0;
+
+	snprintf(buf, PATH_MAX - 1, "/proc/irq/%i/smp_affinity", irq);
+	file = fopen(buf, "r+");
+	if (!file)
+		return -1;
+
```

Comment:
Failure to open the file as r+ could mean many things.  It could mean you're running irqbalance as non-root users.  You need to be careful with this.